shakenfist/virt.py

```diff
@@ -616,27 +582,30 @@ def _create_domain_xml(self):
         # NOTE(mikal): the database stores memory allocations in MB, but the
         # domain XML takes them in KB. That wouldn't be worth a comment here if
         # I hadn't spent _ages_ finding a bug related to it.
+        block_devices = db.get_instance_attribute(
+            self.static_values['uuid'], 'block_devices')
+        ports = db.get_instance_attribute(self.static_values['uuid'], 'ports')
         xml = t.render(
-            uuid=self.uuid,
-            memory=self.memory * 1024,
-            vcpus=self.cpus,
-            disks=self.block_devices['devices'],
+            uuid=self.static_values['uuid'],
```

Comment:
It would be nice to make this code fragment `static_values['uuid']` simpler and less likely to be mistyped.

A possible source of errors is unchecked constants like `'uuid'`. 

(We would never make a typo but others might.)